server/service/apple_mdm.go

```diff
@@ -3180,6 +3180,10 @@ func (svc *MDMAppleDDMService) DeclarativeManagement(r *mdm.Request, dm *mdm.Dec
 	}
 	level.Debug(svc.logger).Log("msg", "ddm request received", "endpoint", dm.Endpoint)
 
+	if err := svc.ds.InsertMDMAppleDDMRequest(r.Context, dm.UDID, dm.Endpoint, string(dm.Data)); err != nil {
+		return nil, err
```

Comment:
could you wrap this error? our DB errors are not very good out of the box.

Reply:
It's wrapped https://github.com/fleetdm/fleet/pull/17844/files#diff-2f8a6d69f21403d6845da98a6a9a7e3808f275a21ec0209cbabcdbeaae57676eR3529 here but I'll add this a wrap here as well